### ISSUE TITLE
zipfile.ZipFile does not create file in fake file system on Windows with Python 2.7.9

### ISSUE THREAD

**Issue by borismod** [jmcgeheeiv/pyfakefs#103]
I am using zipfile package for creating zip file. The below test fails. Is there any way to make it work?

```Python
import os
import zipfile
from pyfakefs import fake_filesystem_unittest


class TestPackageBuilder(fake_filesystem_unittest.TestCase):
    def setUp(self):
        self.setUpPyfakefs()
   
    def test_zip(self):
        self.fs.CreateFile(u'C:\\work\\readme.txt', contents='')
        zip_file_path = os.path.join(u'c:\\work', 'some_file.zip')
        with zipfile.ZipFile(zip_file_path, 'w') as zip_file:
            zip_file.write(u'C:\\work\\readme.txt')

        self.assertFileExists(u'c:\\work\\some_file.zip')

    def assertFileExists(self, file_path):
        self.assertTrue(os.path.exists(file_path), msg='File {0} does not exist'.format(file_path))

```

**Comment by mrbean-bremen** [comment]
This may be an issue with case-sensitivity, which is fixed in the master branch, but not yet in the PyPi package. Try to use the same case everywhere (there is both 'C:' and 'c:' used), or try to install the master branch:
pip install -U https://github.com/jmcgeheeiv/pyfakefs/archive/master.zip
and check if it works.

**Comment by borismod** [comment]
Still fails with master and with same case:

```Python
    def test_zip(self):
        self.fs.CreateFile(u'c:\\work\\readme.txt', contents='')
        zip_file_path = os.path.join(u'c:\\work', 'some_file.zip')
        with zipfile.ZipFile(zip_file_path, 'w') as zip_file:
            zip_file.write(u'c:\\work\\readme.txt')

        self.assertFileExists(u'c:\\work\\some_file.zip')
```

**Comment by mrbean-bremen** [comment]
That's strange... I just tried it locally (with the master branch) and it works fine for me (both versions).

**Comment by borismod** [comment]
I am running on Win10, Pyton 2.7 if it matters

**Comment by mrbean-bremen** [comment]
I am currently run under Python 2.7 and Windows 7 (at work), which should not matter, but I may test it at Windows 10 at home in the evening. Do you use a specific test runner or environment?

**Comment by borismod** [comment]
Nope, I am running on PyCharm Pro 5.0.4, which should not matter

**Comment by mrbean-bremen** [comment]
Ok, I use PyCharm, too - I may have a look at it at later under Windows 10. No more ideas at the moment.

**Comment by mrbean-bremen** [comment]
Ok, same result:

package_test.py:
```
import unittest
...
<your test code>

if __name__ == '__main__':
    unittest.main()

```        
Gives:
```
Microsoft Windows [Version 10.0.10586]

C:\dev\test>python --version
Python 2.7.11

C:\dev\test>python package_test.py
.
----------------------------------------------------------------------
Ran 1 test in 0.003s

OK

```
Van you verify that it fails for you in this same setup?

**Comment by borismod** [comment]
Thanks. I'll check it on Sunday.

**Comment by borismod** [comment]
I fixed the case sensitivity issue in my tests and also installed the master.zip. Still fails. The strange thing is that the an actual zip file is created in the real file system and os.walk on the Fake FS returns only the text file.  Here is my code:

```Python
import os
import zipfile
from pyfakefs import fake_filesystem_unittest


class TestPackageBuilder(fake_filesystem_unittest.TestCase):
    def setUp(self):
        self.setUpPyfakefs()

    def test_zip(self):
        self.fs.CreateFile('c:\\work\\readme.txt', contents='content')
        with zipfile.ZipFile('c:\\work\\some_file.zip', 'w') as zip_file:
            zip_file.write('c:\\work\\readme.txt')

        for f in os.walk('c:\\work'):
            print f

        self.assertFileExists('c:\\work\\some_file.zip')

    def assertFileExists(self, file_path):
        self.assertTrue(os.path.exists(file_path), msg='File {0} does not exist'.format(file_path))
```

**Comment by mrbean-bremen** [comment]
Ok, sounds like zipfile doesn't use the fake filesystem for some reason. I just tried your version on the command line (adding import unittest and main, as shown above), and the output is:
```
C:\...\test>python package_test.py
('c:\\work', [], ['readme.txt', 'some_file.zip'])
.
----------------------------------------------------------------------
Ran 1 test in 0.031s
OK

```
Are you sure there is no other code involved? Have you tried the bare command line test?

**Comment by borismod** [comment]
Fails also from the command line:
![image](https://cloud.githubusercontent.com/assets/209971/15273950/418e7e90-1ab0-11e6-88e7-dc3b4bd4d1e8.png)

**Comment by mrbean-bremen** [comment]
Ok, I'm at loss here. zipfile uses open(), and that should be stubbed out to use fake_filesystem.FakeFileOpen, which it does for me. Maybe jmcgeheeiv (the owner of pyfakefs) has an idea, I cannot think of something at the moment... I guess you are using the latest Python 2.7 (just to be sure to have the same zipfile implementation).

**Comment by borismod** [comment]
OMG, upgrading from Python 2.7.9 to 2.7.10 fixed the problem. 

Thanks a lot for your patience!

**Comment by mrbean-bremen** [comment]
Didn't expect it, but well - you are welcome!

**Comment by mrbean-bremen** [comment]
Seems to be the same problem as issue #70 - zipfile used io.open() up to Python 2.7.9, and open() from 2.7.10 on.

**Comment by jmcgeheeiv** [comment]
Fixed by PR #109
